### ISSUE TITLE
django-cms 3.4.1 : django.core.exceptions.ImproperlyConfigured: A BootstrapColumnPlugin requires a valid parent

### ISSUE THREAD

**Issue by pcolmant** [jrief/djangocms-cascade#202]
Hi Jacob,

It seems tha this issue is due to 

```
@property
def parent(self):
    return self.cms_plugin_instance.parent
```

which has been removed from the class ``CMSPluginBase`` of Django CMS 3.4.x.
It was still present in Django CMS 3.3.x

The ``self.cms_plugin_instance`` still exist, but it's value is set to ``None`` and then never change any more in Django CMS. 

The parent property is used by ``get_parent_instance(self, request=None)`` of the class ``CascadePluginBase``.

As a work around wich works both in CMS 3.3.x and 3.4.x, may be, if any better, to replace into plugin_base.py : 

```
def get_parent_instance(self, request=None):        
        """
        Get the parent model instance corresponding to this plugin. When adding a new plugin, the
        parent might not be available. Therefore as fallback, pass in the request object.
        """                
        try:
            parent_id = self.parent.id
        except AttributeError:
            parent_id = request.GET.get('plugin_parent') if request else None
        for model in CascadeModelBase._get_cascade_elements():
            try:
                return model.objects.get(id=parent_id)
            except model.DoesNotExist:
                continue
```

By : 

```
    def get_parent_instance(self, request=None):
        """
        Get the parent model instance corresponding to this plugin. When adding a new plugin, the
        parent might not be available. Therefore as fallback, pass in the request object.
        """
        if request:
            parent_id = request.GET.get('plugin_parent', None)
            if parent_id is None:
                from cms.models import CMSPlugin
                parent_id = CMSPlugin.objects.filter(id=request.resolver_match.args[0]
                                                     ).only("parent_id").order_by('?').first().parent_id
        else:
            parent_id = None
        for model in CascadeModelBase._get_cascade_elements():
            try:
                return model.objects.get(id=parent_id)
            except model.DoesNotExist:
                continue
```

Sorry to not have made a PR.
Thks for your plugins !

Patrick

**Comment by rfleschenberg** [comment]
Thanks for the report. Can you give instructions on how to reproduce the problem?

**Comment by pcolmant** [comment]
Hi René,

Here are the steps.
I hope this is what you wait...

1/
![image](https://cloud.githubusercontent.com/assets/488513/20106668/41278936-a5d6-11e6-93dd-3e8a57b23f4d.png)
2/
![image](https://cloud.githubusercontent.com/assets/488513/20106706/61edd54e-a5d6-11e6-81c0-a105cb349fdc.png)
3/
![image](https://cloud.githubusercontent.com/assets/488513/20106725/79d5c4dc-a5d6-11e6-823c-5fda3f5c374f.png)
4/
![image](https://cloud.githubusercontent.com/assets/488513/20106774/9a2e0078-a5d6-11e6-93a0-69df7d580dfc.png)

The requirements :  
Django==1.9.11
Pillow==3.4.2
Unidecode==0.04.19
aldryn-reversion==1.0.9
argparse==1.2.1
cmsplugin-filer==1.1.3
diff-match-patch==20121119
django-appconf==1.0.2
django-classy-tags==0.8.0
django-cms==3.4.1
# django-cms==3.3.0
django-compressor==2.1
django-crispy-forms==1.6.1
django-easy-select2==1.3.3
django-filer==1.2.5
django-formtools==1.0
django-import-export==0.5.1
django-mptt==0.8.6
django-mptt-admin==0.4.1
django-parler==1.6.5
django-polymorphic==1.0.2
django-reversion==1.10.2
django-sekizai==0.10.0
django-treebeard==4.0.1
djangocms-admin-style==0.2.8
# djangocms-attributes-field==0.1.1
djangocms-cascade==0.11.0
djangocms-text-ckeditor==3.0.1
djangorestframework==3.4.7
easy-thumbnails==2.3
html5lib==0.9999999
jsonfield==1.0.3
openpyxl==1.8.6
psycopg2==2.6.2
rcssmin==1.0.6
rjsmin==1.0.12
six==1.10.0
tablib==0.11.2
# tzlocal==1.3
uWSGI==2.0.14
wsgiref==0.1.2

**Comment by jrief** [comment]
@pcolmant just for curiosity, does it work with django-cms==3.3.3 ?

**Comment by sostrowsk** [comment]
@jrief 
I'll get the same error in 3.4.1
With 3.3.3 it works fine.

**Comment by pcolmant** [comment]
Hi Jacob,

The change of code in django cms has been done between v3.3.x and 3.4.x.

3.3.3 : 
https://github.com/divio/django-cms/blob/3.3.3/cms/plugin_base.py
The def parent(self): is still present
3.4.0 : 
https://github.com/divio/django-cms/blob/3.4.0/cms/plugin_base.py
1/ The def parent(self): is removed
2/ "self.cms_plugin_instance" still exist, but it's value is set to "None" and then never change any more in Django CMS.

To be sure, I have tried with django-cms==3.3.3 in requirement.txt, pip install -r requierement.txt, reload uwsgi.

So with 3.3.3 it's perfectly fine.
No erreor
Which is not the case with the django-cms 3.4.x branch where some code has been removed...

![image](https://cloud.githubusercontent.com/assets/488513/20109909/6f8a4032-a5e2-11e6-838e-350e34b07fab.png)

Kr,

Patrick
